src/Data/DivisionRing.purs

```diff
@@ -0,0 +1,55 @@
+module Data.DivisionRing
+  ( class DivisionRing
+  , recip
+  , leftDiv
+  , rightDiv
+  , module Data.Ring
+  , module Data.Semiring
+  ) where
+
+import Data.Ring (class Ring, negate, sub)
+import Data.Semiring (class Semiring, add, mul, one, zero, (*), (+))
+import Data.EuclideanRing ((/))
+
+-- | The `DivisionRing` class is for non-zero rings in which every non-zero
+-- | element has a multiplicative inverse. Division rings are sometimes also
+-- | called *skew fields*.
+-- |
+-- | Instances must satisfy the following laws in addition to the `Ring` laws:
+-- |
+-- | - Non-zero ring: `one /= zero`
+-- | - Non-zero multiplicative inverse: `recip a * a = a * recip a = one` for
+-- |   all non-zero `a`
+-- |
+-- | The result of `recip zero` is left undefined; individual instances may
+-- | choose how to handle this case.
+-- |
+-- | If a type has both `DivisionRing` and `CommutativeRing` instances, then
+-- | it is a field and should have a `Field` instance.
+class Ring a <= DivisionRing a where
+  recip :: a -> a
```

Comment:
I haven't actually, but now that you mention it: I think if we had  `leftDiv` and `rightDiv` as class members, then `recip` would have to be defined as either `recip x = leftDiv one x` or `recip x = rightDiv one x`, and we'd have a risk of having the two not match up properly. Also I imagine it would be easy to mix them up if people had to declare their own instances! I think `recip` is a bit easier to explain too.